mlflow/server/auth/__init__.py

```diff
@@ -1443,6 +1445,191 @@ def delete_scorer_permission():
+
+        if field_name not in self.PROTECTED_FIELDS:
+            return next(root, info, **args)
+
+        try:
+            authorization = authenticate_request()
+            if isinstance(authorization, Response):
+                return None
+            username = authorization.username
+
+            if store.get_user(username).is_admin:
+                return next(root, info, **args)
+        except Exception:
+            _logger.warning("GraphQL authorization failed: auth system error", exc_info=True)
+            return None
+
+        try:
+            if not self._check_authorization(field_name, args, username):
+                _logger.debug(f"GraphQL authorization denied for {field_name} by user {username}")
+                return None
+        except MlflowException as e:
+            if e.error_code == ErrorCode.Name(RESOURCE_DOES_NOT_EXIST):
+                return None
+            else:
+                _logger.warning(f"GraphQL authorization error for {field_name}: {e}")
+                return None
+        except Exception:
+            _logger.warning(f"GraphQL authorization error for {field_name}", exc_info=True)
+            return None
+
+        return next(root, info, **args)
+
+    def _check_authorization(self, field_name: str, args: dict[str, Any], username: str) -> bool:
+        """
+        Check if the user is authorized to access the requested field.
+
+        Args:
+            field_name: The GraphQL field being resolved
+            args: The field arguments
+            username: The authenticated username
+
+        Returns:
+            True if authorized, False otherwise
+        """
+        input_obj = args.get("input")
+        if input_obj is None:
+            # No input means no specific resource to check
+            return True
+
+        if field_name == "mlflowGetExperiment":
+            if experiment_id := getattr(input_obj, "experiment_id", None):
+                return _graphql_can_read_experiment(experiment_id, username)
+
+        elif field_name in ("mlflowGetRun", "mlflowListArtifacts"):
+            if run_id := (
+                getattr(input_obj, "run_id", None) or getattr(input_obj, "run_uuid", None)
+            ):
+                return _graphql_can_read_run(run_id, username)
+
+        elif field_name == "mlflowGetMetricHistoryBulkInterval":
+            run_ids = getattr(input_obj, "run_ids", None) or []
+            for run_id in run_ids:
+                if not _graphql_can_read_run(run_id, username):
+                    return False
+
+        elif field_name == "mlflowSearchRuns":
+            if experiment_ids := (getattr(input_obj, "experiment_ids", None) or []):
+                readable_ids = [
+                    exp_id
+                    for exp_id in experiment_ids
+                    if _graphql_can_read_experiment(exp_id, username)
+                ]
+                if not readable_ids:
+                    return False
```

Comment:
This filters the experiment ids, do we need to warn for the invalid experiment ids?